.github/workflows/build-binaries.yml

```diff
@@ -0,0 +1,181 @@
+name: Build Binaries
+on:
+  workflow_dispatch:
+  workflow_call:
+env:
+  CGO_ENABLED: 1
+jobs:
+  get-version:
```

Comment:
I also tried to get the version just once, but unfortunately it doesn't work. GITHUB_OUTPUT is not shared across jobs, just across steps in a single job...
You would need to write the version into a file in get-version job and store that as an artifact, then pull that artifact in each job again and read the file. Also every job reading the version would need to be dependent on get-version to ensure it is run before. After I figured that out I swallowed the pill of calling git describe in every job.